pkg/services/sqlstore/migrations/usermig/user_lowercase_login_and_email.go

```diff
@@ -0,0 +1,90 @@
+package usermig
+
+import (
+	"strings"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+	"github.com/grafana/grafana/pkg/services/user"
+	"xorm.io/xorm"
+)
+
+const (
+	LowerCaseUserLoginAndEmail = "update login and email fields to lowercase"
+)
+
+// Service accounts login were not unique per org. this migration is part of making it unique per org
+// to be able to create service accounts that are unique per org
+func AddLowerCaseUserLoginAndEmail(mg *migrator.Migrator) {
+	mg.AddMigration(LowerCaseUserLoginAndEmail, &UsersLowerCaseLoginAndEmail{})
+}
+
+var _ migrator.CodeMigration = new(UsersLowerCaseLoginAndEmail)
+
+type UsersLowerCaseLoginAndEmail struct {
+	migrator.MigrationBase
+}
+
+func (p *UsersLowerCaseLoginAndEmail) SQL(dialect migrator.Dialect) string {
+	return "code migration"
+}
+
+func (p *UsersLowerCaseLoginAndEmail) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
+	// Get all users
+	users := make([]*user.User, 0)
+	err := sess.Table("user").Find(&users)
+	if err != nil {
+		return err
+	}
+
+	for _, usr := range users {
+		/*
+			LOGIN
+		*/
+		lowerLogin := strings.ToLower(usr.Login)
```

Comment:
thought: if user's login is already lower-cased, then there's no work to be done, and you could just skip on to the email?

Reply:
yea, i guess we could do that. you want to avoid the db calls? this might be good for the big instances we have.

Reply:
il do the same with the email